### ISSUE TITLE
2021Q1 - Migrate wpt-sync's CI from Travis to GitHub Actions

### ISSUE THREAD

**Issue by denschub** [mozilla/webcompat-team-okrs#180]
With travis-ci.org [shutting down at the end of the year](https://docs.travis-ci.com/user/migrate/open-source-repository-migration#frequently-asked-questions) and travis-ci.com [limiting the amount of CI time FOSS projects can have](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing), we should move the wpt-sync repo from Travis to GitHub Actions.

There is no really complex build matrix for that project, so migration should be relatively simple.

**Comment by karlcow** [comment]
Seen this on slack at Mozilla. 

> I’ve been porting Jazzband (https://jazzband.co/) projects over to GitHub Actions in the past weeks and it’s been surprisingly simple since there is a tox-gh-actions integration that is similar to tox-travis

So I look for an example in their projects and found:

https://github.com/jazzband/django-configurations/commit/78f9824f0c7959bec01e5caf6582fbe7cb32cbd1

**Comment by denschub** [comment]
I have this working in a private fork of mine, but running into occasional timeouts and GItHub support is currently looking into why (because it makes no sense!)

**Comment by denschub** [comment]
This actually is... blocked.

I have created [a PR](https://github.com/mozilla/wpt-sync/pull/887) with the configs that appear to work, but we can't actually ship this yet. See https://bugzilla.mozilla.org/show_bug.cgi?id=1594423 (MoCo-internal) for details...

**Comment by denschub** [comment]
Based on [this bug](https://github.com/mozilla/wpt-sync/issues/884), I've moved this to "High priority". This is also no longer blocked, because GitHub Actions are now available for mozilla projects as long as they do not require external actions (i.e. "steps" that are not provided by GitHub).

During my work last year, I discovered that there was a failing test. This turned out to be an issue with non-deterministic ordering of browser results, causing a matching-string test to fail. I addressed that in the [existing PR](https://github.com/mozilla/wpt-sync/pull/887), and with that, that PR is now ready to be landed.
